main.c

```diff
@@ -1,11 +1,97 @@
 #include <stdbool.h>
 #include <stdio.h>
 #include <string.h>
+#include <unistd.h>
 #include "mzcc.h"
 
+char *outfile = NULL, *infile = NULL;
+extern FILE *outfp;
+bool dump_ast;
+
+static void usage()
+{
+    fprintf(stdout,
+            "\n"
+            "Usage : mzcc <input file>\n"
+            "\n"
+
+            "  -o filename     Output to the specified file\n"
+            "  -dump-ast       Print Abstract Syntax Tree\n\n");
+}
+void print_usage_and_exit()
+{
+    usage();
+    exit(1);
+}
+static void parseopt(int argc, char **argv)
+{
+    if (argc < 2) {
+        print_usage_and_exit();
+    }
+
+    while (true) {
+        argc--;
+        argv++;
+        if (argc == 0) {
+            break;
+        }
+
+        if ((*argv)[0] == '-') {
+            switch ((*argv)[1]) {
+            case 'o':
+                argc--;
+                argv++;
+                outfile = *argv;
+                break;
+            case 'd':
+                if (strcmp(*argv, "-dump-ast")) {
+                    print_usage_and_exit();
+                } else {
+                    dump_ast = true;
+                    break;
+                }
+            default:
+                print_usage_and_exit();
+            }
+        } else {
+            if (infile == NULL) {
+                infile = argv[0];
+            } else {
+                // The second non-option argument is not what we expect.
+                print_usage_and_exit();
+            }
+        }
+    }
+}
+static FILE *open_output_file()
+{
+    FILE *tmp = stdout;
+    if (outfile != NULL && ((tmp = fopen(outfile, "w")) == NULL)) {
+        printf("Can not open file %s\n", outfile);
+        exit(1);
+    }
+    return tmp;
+}
+static FILE *open_input_file()
+{
+    if (infile == NULL) {
+        printf("Input file is not specified\n\n");
+        print_usage_and_exit();
+    }
+
+    FILE *tmp = fopen(infile, "r");
+    if (tmp == NULL) {
+        printf("Can not open file %s\n", infile);
+        exit(1);
+    }
+    return tmp;
+}
```

Comment:
Ditto. New line.

Reply:
Do you mean appending a new line after the right braces ?

Reply:
Exactly.